File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int finalposition=nums.length-1;
        int n=nums.length;
        for(int i=n-2;i>=0;i--){
            if(i+nums[i]>=finalposition){
                finalposition=i;
            }
        }
        return finalposition==0;
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n=len(nums)
        finalpos=n-1
        for i in range(n-2,-1,-1):
            if(i+nums[i]>=finalpos):
                finalpos=i
        return finalpos==0

        